module/applications/calendar/calendar-hud.mjs

```diff
@@ -0,0 +1,280 @@
+            amount: formatTime(value, unit).titleCase()
+          })
+        }))
+      },
+      {
+        action: "openPartySheet",
+        icon: "fa-solid fa-users",
+        position: "end",
+        tooltip: game.i18n.localize("DND5E.CALENDAR.Action.OpenPartySheet"),
+        visible: game.settings.get("dnd5e", "primaryParty")?.actor
+          ?.testUserPermission(game.user, CONST.DOCUMENT_OWNERSHIP_LEVELS.LIMITED)
+      }
+    ];
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _prepareContext(options) {
+    const context = await super._prepareContext(options);
+    this._prepareButtonsContext(context, options);
+    return context;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare the buttons that can be displayed around the calendar UI.
+   * @param {ApplicationRenderContext} context  Context being prepared.
+   * @param {HandlebarsRenderOptions} options   Options which configure application rendering behavior.
+   */
+  async _prepareButtonsContext(context, options) {
+    /**
+     * A hook event that fires when preparing the buttons displayed around the calendar HUD. Buttons in each list
+     * are sorted with those closest to the center first.
+     * @function dnd5e.prepareCalendarButtons
+     * @memberof hookEvents
+     * @param {CalendarHUD} app              The Calendar HUD application being rendered.
+     * @param {CalendarHUDButton[]} buttons  Buttons displayed around the calendar UI.
+     */
+    const controls = this._doEvent(this._getCalendarButtons, {
+      async: false,
+      debugText: "Calendar Control Buttons",
+      hookName: "dnd5e.prepareCalendarButtons",
+      hookResponse: true,
+      parentClassHooks: false
+    });
+
+    const prepareCalendarButton = (data, index, parent) => ({
+      ...data, index,
+      additional: data.additional ? data.additional.map((a, i) => prepareCalendarButton(a, i, data)) : undefined,
+      tooltipDirection: (parent?.position ?? data.position) === "start" ? "LEFT" : "RIGHT"
+    });
+
+    context.buttons = controls
+      .filter(b => typeof b.visible === "function" ? b.visible.call(this) : b.visible ?? true)
+      .map(prepareCalendarButton);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _preparePartContext(partId, context, options) {
+    context = await super._preparePartContext(partId, context, options);
+    switch ( partId ) {
+      case "endButtons":
+        context.buttons = context.buttons.filter(b => b.position === "end");
+        break;
+      case "startButtons":
+        context.buttons = context.buttons.filter(b => b.position === "start").reverse();
+        break;
+    }
+    return context;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Adjust the date, time, and progress in the core part without a full re-render to allow animation.
```

Comment:
Are we preventing `core` from being rendered via the normal pipeline somewhere? Otherwise I'd have thought you'd need something like this:

```js
_configureRenderOptions(options) {
  super._configureRenderOptions(options);
  if ( this.rendered ) options.parts = options.parts.filter(p => p !== "core");
}
```

Reply:
Well this is just for when the time is updated, in which case the hook calls `renderCore` directly. The only other time we call render directly is first render and settings updates, both of which shouldn't care about keeping the animations intact. That said, I don't think this would hurt anything.